### PR TITLE
bump github workflows go version

### PR DESCRIPTION
**Pull request by gabo1208**
Bump go versions in github release workflow + kinde2e workflow.

Using go 1.15.x (as it is now)
image

Using go 1.17.x
image

Almost a 10% reduced size on Sources files and probably more performant code

### CODE REVIEW

File: .github/workflows/release.yaml [hunk 1]
```diff
@@ -58,7 +58,7 @@ jobs:
 
     - name: Get Release URL
       id: get_release_url
-      uses: bruceadams/get-release@v1.2.0
+      uses: bruceadams/get-release@v1.2.3
```
Comment (xtreme-sameer-vohra) [inline .github/workflows/release.yaml:61]: nit: to keep changes atomic, would be good to put this in a separate PR

File: .github/workflows/kind-e2e.yaml [hunk 2]
```diff
@@ -17,14 +17,14 @@ jobs:
...
+    - name: Set up Go 1.17.x
       uses: actions/setup-go@v2
       with:
-        go-version: 1.16.x
+        go-version: 1.17.x
 
     - name: Install Dependencies
       env:
-        KO_VERSION: 0.8.3
+        KO_VERSION: 0.9.3
```
Comment (xtreme-sameer-vohra) [inline .github/workflows/kind-e2e.yaml:27]: Nice, good to synchronize the version across all uses of ko
nit: keep changes atomic and split the `go` bump and `ko` bump

File: .github/workflows/kind-e2e.yaml [hunk 3]
```diff
@@ -17,14 +17,14 @@ jobs:
       KO_DOCKER_REPO: kind.local
 
     steps:
-    - name: Set up Go 1.16.x
+    - name: Set up Go 1.17.x
```
Comment (xtreme-sameer-vohra) [inline .github/workflows/kind-e2e.yaml:20]: Nice, good to synchronize the version across all uses go

**Review by xtreme-sameer-vohra — CHANGES_REQUESTED** [review]

File: .github/workflows/kind-e2e.yaml (hunk 2 above)
Comment (embano1) [reply .github/workflows/kind-e2e.yaml:27]: shall we use the ko action instead of curling?

Comment (gabo1208) [reply .github/workflows/kind-e2e.yaml:27]: sounds cool! kk let me try this in the next pr

**Comment by embano1** [comment]
Can you please update the title/description of this PR as `ko` is not touched here at all.

**Review by embano1 — APPROVED** [review]
LGTM
